src/data/literal.rs

```diff
@@ -393,4 +394,164 @@ mod tests {
         matches!(Null.modulo(&num_divisor("2")).unwrap(), Null);
         matches!(Null.modulo(&Null).unwrap(), Null);
     }
+    #[test]
+    fn partial_eq() {
+        use crate::data::interval::Interval as I;
+        macro_rules! text {
+            ($text: expr) => {
+                Text(Cow::Owned($text.to_owned()))
+            };
+        }
+        macro_rules! itv {
+            ($itv: expr) => {
+                Interval(I::Microsecond($itv))
+            };
+        }
+        macro_rules! num {
+            ($num: expr) => {
+                Number(Cow::Owned($num.to_owned()))
+            };
+        }
+        //Boolean
+        assert_eq!(
+            &Boolean(PartialEq::eq(&Boolean(true), &Boolean(true))),
+            &Boolean(true)
+        );
```

Comment:
You don't need to use `Literal::Boolean` and `PartialEq::eq`.
`Literal::Boolean` wrapping is unnecessary.
`PartialEq::eq` is called implicitly by `==` comparison.
Therefore, this would be enough.
```rust
assert_eq!(Boolean(true), Boolean(true))
```

You can apply this to the codes below.

Reply:
![image](https://user-images.githubusercontent.com/22141521/131247402-f7028dbd-edba-4187-b26b-be7be152bcd2.png)
i don't know how to resolve this. Without casting to Boolean, i got this error

Reply:
You can simply follow `clippy`'s suggestion.

```rust
assert_eq!(Boolean(true), Boolean(true));
```
or
```rust
assert!(Boolean(true) == Boolean(true));
```

This clippy error is because you tried using `assert_eq!` macro for comparing literal boolean values.

Reply:
Finally i understood!! thanks